crates/re_space_view_dataframe/src/space_view_class.rs

```diff
@@ -0,0 +1,190 @@
+
+    fn icon(&self) -> &'static re_ui::Icon {
+        //TODO(ab): fix that icon
+        &re_ui::icons::SPACE_VIEW_TEXTBOX
+    }
+
+    fn help_text(&self, _re_ui: &re_ui::ReUi) -> egui::WidgetText {
+        "Show the data contained in entities in a table.".into()
+    }
+
+    fn on_register(
+        &self,
+        system_registry: &mut re_viewer_context::SpaceViewSystemRegistry,
+    ) -> Result<(), SpaceViewClassRegistryError> {
+        system_registry.register_part_system::<EmptySystem>()
+    }
+
+    fn preferred_tile_aspect_ratio(&self, _state: &Self::State) -> Option<f32> {
+        None
+    }
+
+    fn layout_priority(&self) -> re_viewer_context::SpaceViewClassLayoutPriority {
+        re_viewer_context::SpaceViewClassLayoutPriority::Low
+    }
+
+    fn auto_spawn_heuristic(
+        &self,
+        _ctx: &ViewerContext<'_>,
+        _space_origin: &EntityPath,
+        _ent_paths: &PerSystemEntities,
+    ) -> re_viewer_context::AutoSpawnHeuristic {
+        AutoSpawnHeuristic::NeverSpawn
+    }
+
+    fn selection_ui(
+        &self,
+        _ctx: &ViewerContext<'_>,
+        _ui: &mut egui::Ui,
+        _state: &mut Self::State,
+        _space_origin: &EntityPath,
+        _space_view_id: SpaceViewId,
+        _root_entity_properties: &mut EntityProperties,
+    ) {
+    }
+
+    fn ui(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        _state: &mut Self::State,
+        _root_entity_properties: &EntityProperties,
+        query: &ViewQuery<'_>,
+        _system_output: SystemExecutionOutput,
+    ) -> Result<(), SpaceViewSystemExecutionError> {
+        re_tracing::profile_function!();
+
+        let entities: Vec<_> = query
+            .iter_all_data_results()
+            .filter(|data_result| data_result.resolved_properties.visible)
+            .map(|data_result| &data_result.entity_path)
+            .unique()
+            .cloned()
+            .collect();
+
+        let store = ctx.store_db.store();
+        let latest_at_query = query.latest_at_query();
+
+        // for each entity, this does the union of all instance keys of all components
+        let all_instances: Vec<_> = entities
+            .iter()
+            .flat_map(|entity| {
+                store
+                    .all_components(&query.timeline, entity)
+                    .unwrap_or_default()
+                    .into_iter()
+                    .filter(|comp| !comp.is_indicator_component())
+                    .flat_map(|comp| {
+                        get_component_with_instances(store, &latest_at_query, entity, comp)
+                            .map(|(_, comp_inst)| comp_inst.instance_keys())
+                            .unwrap_or_default()
+                    })
+                    .filter(|instance_key| !instance_key.is_splat())
+                    .map(|instance_key| InstancePath::instance(entity.clone(), instance_key))
+            })
+            .unique()
+            .collect();
```

Comment:
It's very hard to tell from this block of code what order we will show all the instances. I would expect to show the table sorted by the instance paths, since that is our primary key (and we can't change which column we sort on).

The outer `unique` seems wrong btw - the `entities` are already unique, so if there is something that needs deduping it should only be in the inner closure.